src/openads-appnexus/repository/PullingDataEntry.js

```diff
@@ -0,0 +1,44 @@
+const TIMEOUT_EXCEPTION = 'Timeout retrieving the Ad from the server'
+
+export default class PullingDataEntry {
+  constructor ({id, adRepository, timeout, wait}) {
+    this._id = id
+    this._adRepository = adRepository
+    this._timeout = timeout
+    this._wait = wait
+    this._dataPromise = Promise.race([
+      this._createIntervalPromise({id}),
+      this._createTimeoutPromise({id})
+    ])
+  }
+
+  get id () {
+    return this._id
+  }
+
+  waitForData () {
```

Comment:
is this really waiting for Data? I mean, the execution of _Promise.race_ was not already triggered by the constructor when _this._dataPromise_ was initialized?

Reply:
Yep, it's waiting :)